add and update blobber separated

Hitenjain14 opened the pull request:
## Fixes
addBlobber() could update blobber too. 
Separated the functionality of both , if any repos are using addBlobber() to update blobber , they should use blobber_update_settings .


## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber: 
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

YarikRevich committed:
feature: added endpoint

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
feature: made 'nonce' argument optional

YarikRevich committed:
feature: modified user for ethereum addresses

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
Merge branch 'staging' of https://github.com/0chain/0chain into feature/client_burns_endpoint

YarikRevich committed:
fix: fixed structure serialization for msgp

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
feature: added tests for burn method

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
feature: added more test cases

YarikRevich committed:
Merge branch 'staging' of https://github.com/0chain/0chain into feature/client_burns_endpoint

YarikRevich committed:
Merge branch 'staging' into feature/client_burns_endpoint

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
Merge branch 'feature/client_burns_endpoint' of https://github.com/0chain/0chain into feature/client_burns_endpoint

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
Update state_handler.go

YarikRevich committed:
fix: fixed marshaling bug

YarikRevich committed:
feature: implemeneted event db writing of mint nonce and burn tickets

YarikRevich committed:
feature: added test environment

YarikRevich committed:
fix: fixed state handler tests

YarikRevich committed:
fix: fixed bridge tests

YarikRevich committed:
Merge branch 'feature/client_burns_endpoint' into feature/reconciliation_problem

YarikRevich committed:
Merge branch 'staging' into feature/reconciliation_problem

YarikRevich committed:
Merge branch 'staging' of https://github.com/0chain/0chain into feature/reconciliation_problem

Hitenjain14 committed:
add and update blobber seperated

Hitenjain14 committed:
add and update blobber seperated

Hitenjain14 committed:
test fix

Hitenjain14 committed:
test fix

YarikRevich committed:
fix: fixed burn ticket tests

YarikRevich committed:
Merge branch 'feature/reconciliation_problem' of https://github.com/0chain/0chain into feature/reconciliation_problem

YarikRevich committed:
Merge https://github.com/0chain/0chain into feature/reconciliation_problem

Hitenjain14 committed:
duplicate test

Hitenjain14 committed:
duplicate test

YarikRevich committed:
fix: fixed burn ticket table tests

YarikRevich committed:
Update db_setup_test.go

YarikRevich committed:
fix: removed dead code

YarikRevich committed:
Merge branch 'feature/reconciliation_problem' of https://github.com/0chain/0chain into feature/reconciliation_problem

YarikRevich committed:
feature: updated migration step to add a proper column

peterlimg commented on code/go/0chain.net/smartcontract/storagesc/blobber_test.go:
```diff
@@ -1322,3 +1298,26 @@ func TestBlobberHealthCheck(t *testing.T) {
+
+	b.BaseURL = "https://newabcurl.com"
+	_, err = updateBlobberAdd(t, b, 0, tp, ssc, balances)
+	require.Error(t, err)
```
Please assert the error message to make sure it returns the expected error.

peterlimg commented:
@Hitenjain14 please do search the add blobber endpoint in https://github.com/0chain/blobber repo to see if it is used for updating, and correct it if any.

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
feature: modified return model for state handler

YarikRevich committed:
Merge branch 'staging' into feature/reconciliation_problem

YarikRevich committed:
Update burn_ticket_test.go

YarikRevich committed:
Merge https://github.com/0chain/0chain into feature/reconciliation_problem

Hitenjain14 commented:
> @Hitenjain14 please do search the add blobber endpoint in https://github.com/0chain/blobber repo to see if it is used for updating, and correct it if any.

I have raised this  [pr](https://github.com/0chain/blobber/pull/961) to address this issue , talked with @dabasov and we shouldn't be doing any updates from blobber server. Should only be done from cli and thats sending `update_blobber_settings` transaction.

Hitenjain14 committed:
equal error msg

Hitenjain14 committed:
equal error msg

YarikRevich committed:
Merge branch 'staging' into feature/reconciliation_problem

YarikRevich committed:
Merge branch 'staging' of https://github.com/0chain/0chain into feature/reconciliation_problem

dabasov commented on code/go/0chain.net/smartcontract/storagesc/helper_test.go:
```diff
@@ -171,6 +172,18 @@ func updateBlobber(t testing.TB, blob *StorageNode, value currency.Coin, now int
 	return ssc.addBlobber(tx, input, balances)
 }
 
+func updateBlobber(t testing.TB, blob *StorageNode, value currency.Coin, now int64,
```
looks like it is duplicate of `updateBlobberAdd`

Hitenjain14 replied:
Yes but in `updateBlobberAdd` we are calling `ssc.addBlobber(tx, input, balances)` , its used to test that we cannot perform any updates using `addBlobber` now

peterlimg replied:
@Hitenjain14 if `updateBlobberAdd` would confuse users, let's rename it to `addBlobber`.

Hitenjain14 replied:
Its a helper test function , using `addblobber` to update the blobber which should fail and thats what we check .

YarikRevich committed:
feature: fixed failing tests

YarikRevich committed:
Merge branch 'feature/reconciliation_problem' of https://github.com/0chain/0chain into feature/reconciliation_problem

YarikRevich committed:
fix: fixed failing tests

YarikRevich committed:
feature: moved 'not_processed_burn_tickets' endpoint to sharders

YarikRevich committed:
Merge branch 'staging' into feature/reconciliation_problem

YarikRevich committed:
feature: moved mint nonce handler to sharder

YarikRevich committed:
Merge branch 'feature/reconciliation_problem' of https://github.com/0chain/0chain into feature/reconciliation_problem

YarikRevich committed:
fix: fixed bugs

YarikRevich committed:
fix: fixed burn ticket nonce error

YarikRevich committed:
fix: fixed bugs

peterlimg commented on code/go/0chain.net/smartcontract/storagesc/stakepool.go:
```diff
@@ -361,6 +361,28 @@ func (ssc *StorageSmartContract) getOrCreateStakePool(
+	providerType spenum.Provider,
+	providerId datastore.Key,
+	settings stakepool.Settings,
+	balances chainstate.StateContextI,
```
providerType, providerId and balances are not being used, let's remove them

dabasov committed:
refactoring
moved handler to sharders
removed access to events db from sc

Hitenjain14 committed:
code review changes

Hitenjain14 committed:
code review changes

YarikRevich committed:
fix: fixed handler tests

YarikRevich committed:
Merge branch 'staging' into feature/reconciliation_problem

peterlimg reviewed (approved):
LGTM

YarikRevich committed:
Merge branch 'staging' into feature/reconciliation_problem

YarikRevich committed:
Merge https://github.com/0chain/0chain into feature/reconciliation_problem

YarikRevich committed:
fix: fixed failing test

YarikRevich committed:
Merge branch 'feature/reconciliation_problem' of https://github.com/0chain/0chain into feature/reconciliation_problem

YarikRevich committed:
Merge branch 'staging' into feature/reconciliation_problem

peterlimg committed:
Optimize payFee txn

- Do not load all miners and sharders for rewarding, load ids and get random ones first, then load selected nodes.
- Do not load all miners and sharders for view change pool work to delete nodes, save node ids that will be deleted and remove them all at once.

peterlimg committed:
Change num sharder rewarded each round from 5 to 1 and

delegate pool from 1 to 5.

Saving sharder node and all delegate pool are slow, so instead of saving 5 nodes and each node reward 1 pool, we'd better save 1 node and each node reward 5 pools.

peterlimg committed:
Clean up commented code

peterlimg committed:
Fix unit tests

peterlimg committed:
Fix incorrect conversion between integer types

closes
- https://github.com/0chain/0chain/security/code-scanning/27
- https://github.com/0chain/0chain/security/code-scanning/26

dabasov committed:
Merge branch 'staging' into blobber-add-update

service-0chain commented:
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | blobber-add-update  |
| blobber  | blobber-server-update  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

dabasov committed:
Merge branch 'staging' into feature/reconciliation_problem

dabasov committed:
deleted not needed file

peterlimg committed:
Merge pull request #2198 from 0chain/optimize-pay-fees

Optimize payFee

peterlimg committed:
Merge pull request #2199 from 0chain/fix/security-alert

Fix security alert

YarikRevich committed:
Merge branch 'staging' into feature/reconciliation_problem

YarikRevich committed:
Merge pull request #2143 from 0chain/feature/reconciliation_problem

Fix: reconciliation problem

Hitenjain14 committed:
Merge branch 'blobber-add-update' of https://github.com/0chain/0chain into blobber-add-update

dabasov reviewed (approved):
lgtm
